### PR TITLE
[C-2089] Distinguish between errored and abandoned

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Distinguishes between errored and abandoned, to make it clearer to us whether a track/collection is not downloadable vs an error in downloading. For now we don't need to do anything with this information, but setting it up now so we can add some tooling around handling these cases separately in the future, ie re-trying errored.

Also attempts to fix drawer glitch and improves collection header loading states.

TODO: Adding abandoned actions, one moment...

### CODE REVIEW

File: packages/mobile/src/utils/trackUtils.ts [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-    Track,
-    'is_available' | 'is_delete' | 'is_invalid' | 'is_unlisted' | 'owner_id'
-  >,
-  currentUserId: number
-) =>
-  track.is_available &&
-  !track.is_delete &&
-  !track.is_invalid &&
-  !(track.is_unlisted && track.owner_id !== currentUserId)
+export const isTrackValid = (track: UserTrackMetadata, currentUserId: ID) => {
```
Comment (amendelsohn) [inline packages/mobile/src/utils/trackUtils.ts:3]: minor, but imo isValid feels pretty generic and to me sounds more like a check for illegal values. Not sure the best name, but I think isAvailableForPlay was actually closer

**Review by amendelsohn — APPROVED** [review]
Nice!

**Review by sliptype — APPROVED** [review]
Makes sense!

File: packages/mobile/src/utils/trackUtils.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/utils/trackUtils.ts:3]: yeah i like that, how about "downloadable"?

**Pull request merged**